generators/counter.py

```diff
@@ -29,25 +29,33 @@ def __init__(self):
 
     def configure(self, context):
         super().configure(context)
-        self._range = range(self.attr_value().start(),
-                            self.attr_value().end() + 1,
-                            self.attr_value().step())
-        self._range_length = len(self._range)
+        if self.attr_value().step() != 0:
+            self._range = range(self.attr_value().start(),
+                                self.attr_value().end() + 1,
+                                self.attr_value().step())
+            self._range_length = len(self._range)
+        else:
+            self._range = self.attr_value().start()
```

Comment:
The fact that this isn't a `range` object will maybe cause issues. I think if you just do this in the case of the 0 step the rest of the block should work as expected.

```python
self._range = range(self.attr_value().start(), self.attr_value().start() + 1, 1)
```